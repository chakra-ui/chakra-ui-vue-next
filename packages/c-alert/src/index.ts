import { defineComponent, computed, createVNode, mergeProps, isVNode, Fragment } from 'vue';
import { createStylesContext, useMultiStyleConfig, chakra } from '@chakra-ui/vue-system';
import { createContext, getValidChildren } from '@chakra-ui/vue-utils';
import { createIconComponent } from '@chakra-ui/c-icon';

function _extends() {
  _extends = Object.assign ? Object.assign.bind() : function (target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];

      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }

    return target;
  };
  return _extends.apply(this, arguments);
}

var CInfoIcon = createIconComponent("info");
var CCheckIcon = createIconComponent("check-circle");
var CWarningIcon = createIconComponent("warning-alt");
var CErrorIcon = createIconComponent("warning");

function _isSlot(s) {
  return typeof s === 'function' || Object.prototype.toString.call(s) === '[object Object]' && !isVNode(s);
}

var STATUSES = {
  info: {
    colorScheme: "blue",
    icon: CInfoIcon
  },
  success: {
    colorScheme: "green",
    icon: CCheckIcon
  },
  warning: {
    colorScheme: "orange",
    icon: CWarningIcon
  },
  error: {
    colorScheme: "red",
    icon: CErrorIcon
  },
  loading: {
    icon: CInfoIcon,
    colorScheme: "blue"
  }
};

var _createStylesContext = createStylesContext("Alert"),
    StylesProvider = _createStylesContext[0],
    useStyles = _createStylesContext[1];

var _createContext = createContext({
  name: "AlertContext",
  errorMessage: "useAlertContext: `context` is undefined. Seems you forgot to wrap alert components in `<c-alert />`"
}),
    AlertProvider = _createContext[0],
    useAlertContext = _createContext[1];
/**
 * CAlert component
 *
 * This is the container component for all Alert components.
 * It also provides state and context to it's compound components
 */


var CAlert = defineComponent({
  name: "CAlert",
  props: {
    as: {
      type: [String, Object],
      "default": "div"
    },
    status: {
      type: [String],
      "default": "info"
    },
    colorScheme: {
      type: [String]
    },
    styleConfig: {
      type: [Object]
    },
    variant: {
      type: [String],
      "default": "solid"
    }
  },
  setup: function setup(props, _ref) {
    var slots = _ref.slots,
        attrs = _ref.attrs;
    var colorScheme = computed(function () {
      var _STATUSES$props$statu;

      return props.colorScheme || ((_STATUSES$props$statu = STATUSES[props == null ? void 0 : props.status]) == null ? void 0 : _STATUSES$props$statu.colorScheme);
    });
    var themingProps = computed(function () {
      return {
        colorScheme: colorScheme.value,
        variant: props.variant
      };
    });
    AlertProvider({
      status: computed(function () {
        return props.status;
      })
    });
    var styles = useMultiStyleConfig("Alert", themingProps);
    StylesProvider(styles);
    var alertStyles = computed(function () {
      return _extends({
        width: "100%",
        display: "flex",
        alignItems: "center",
        position: "relative",
        overflow: "hidden"
      }, styles.value.container);
    });
    return function () {
      return createVNode(chakra.div, mergeProps({
        "role": "alert",
        "__label": "alert",
        "__css": alertStyles.value
      }, attrs), {
        "default": function _default() {
          return getValidChildren(slots);
        }
      });
    };
  }
});
/**
 * CAlertTitle component
 *
 * The title component for alerts
 */

var CAlertTitle = defineComponent({
  name: "CAlertTitle",
  setup: function setup(_, _ref2) {
    var attrs = _ref2.attrs,
        slots = _ref2.slots;
    var styles = useStyles();
    return function () {
      return createVNode(chakra.div, mergeProps({
        "__label": "alert__title",
        "__css": styles.value.title
      }, attrs), _isSlot(slots) ? slots : {
        "default": function _default() {
          return [slots];
        }
      });
    };
  }
});
/**Â«
 * CAlertDescription component
 *
 * The description component for alerts
 */

var CAlertDescription = defineComponent({
  name: "CAlertDescription",
  setup: function setup(_, _ref3) {
    var attrs = _ref3.attrs,
        slots = _ref3.slots;
    var styles = useStyles();
    return function () {
      return createVNode(chakra.div, mergeProps({
        "__label": "alert__description",
        "__css": styles.value.description
      }, attrs), {
        "default": function _default() {
          return getValidChildren(slots);
        }
      });
    };
  }
});
/**
 * CAlertIcon component
 *
 * The Icon component for alerts
 */

var CAlertIcon = defineComponent({
  name: "CAlertIcon",
  setup: function setup(_, _ref4) {
    var attrs = _ref4.attrs,
        slots = _ref4.slots;
    var styles = useStyles();

    var _useAlertContext = useAlertContext(),
        status = _useAlertContext.status;

    var BaseIcon = STATUSES[status.value].icon;
    var css = computed(function () {
      return status.value === "loading" ? styles.value.spinner : styles.value.icon;
    });
    return function () {
      var validChildren = getValidChildren(slots);
      return createVNode(chakra.span, mergeProps({
        "display": "inherit",
        "__label": "alert__icon"
      }, attrs, {
        "__css": css.value
      }), {
        "default": function _default() {
          return createVNode(Fragment, null, [validChildren.length ? slots : createVNode(BaseIcon, {
            "h": "100%",
            "w": "100%"
          }, null)]);
        }
      }); // return <icon {...styles.value.icon} {...attrs}></icon>
    };
  }
});

export { CAlert, CAlertDescription, CAlertIcon, CAlertTitle };
