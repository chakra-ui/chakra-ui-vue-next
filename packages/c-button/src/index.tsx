import { defineComponent } from "vue"

export { default as CButton } from "./button"
export { default as CButtonGroup } from "./button-group"
export { default as CIconButton } from "./icon-button"
