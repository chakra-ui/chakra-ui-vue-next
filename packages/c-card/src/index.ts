export { CCard, type CCardProps } from "./c-card"
export { CCardBody, type CCardBodyProps } from "./c-card-body"
export { CCardHeader, type CCardHeaderProps } from "./c-card-header"
