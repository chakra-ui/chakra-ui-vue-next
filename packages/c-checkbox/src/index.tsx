export { CCheckbox } from "./checkbox"
export {
  CCheckboxGroup,
  useCheckboxGroupContext,
  type CheckboxGroupContext,
} from "./checkbox-group"

export type { CCheckboxControlProps, CCheckboxProps } from "./checkbox"
export type { CCheckboxGroupProps } from "./checkbox-group"
