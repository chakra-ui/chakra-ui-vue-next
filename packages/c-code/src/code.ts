import { h, defineComponent, PropType, computed } from 'vue'
import {
  chakra,
  DOMElements,
  ThemingProps,
  useStyleConfig,
} from '@chakra-ui/vue-system'
import { filterUndefined } from '@chakra-ui/utils'

const CCode = defineComponent({
  props: {
    as: {
      type: [Object, String] as PropType<DOMElements>,
      default: 'div',
    },
    colorScheme: String as PropType<ThemingProps['colorScheme']>,
    styleConfig: String as PropType<ThemingProps['styleConfig']>,
  },
  setup(props, { slots, attrs }) {
    return () => {
      const themingProps = computed<ThemingProps>(() =>
        filterUndefined({
          colorScheme: props.colorScheme,
          styleConfig: props.styleConfig,
        })
      )
      const styles = useStyleConfig('Code', themingProps.value)

      return h(
        chakra(props.as),
        {
          __css: {
            display: 'inline-block',
            verticalAlign: 'middle',
            fontSize: 'sm',
            px: '0.2em',
            fontFamily: 'mono',
            rounded: 'sm',
            ...styles.value,
          },
          ...attrs,
        },
        slots
      )
    }
  },
})

export default CCode
