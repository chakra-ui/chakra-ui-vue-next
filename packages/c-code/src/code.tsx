import { getValidChildren, vueThemingProps } from "@chakra-ui/vue-utils"
import { h, defineComponent, PropType, computed } from "vue"
import {
  chakra,
  DOMElements,
  ThemingProps,
  useStyleConfig,
} from "@chakra-ui/vue-system"
import { filterUndefined } from "@chakra-ui/utils"

const CCode = defineComponent({
  props: {
    as: {
      type: [Object, String] as PropType<DOMElements>,
      default: "code",
    },
    ...vueThemingProps,
  },
  setup(props, { slots, attrs }) {
    const themingProps = computed<ThemingProps>(() =>
      filterUndefined({
        colorScheme: props.colorScheme,
        variant: props.variant,
        size: props.size,
        styleConfig: props.styleConfig,
      })
    )
    const styles = useStyleConfig("Code", themingProps)

    return () => {
      return (
        <chakra.code
          as={props.as}
          __css={{
            display: "inline-block",
            verticalAlign: "middle",
            fontSize: "sm",
            px: "0.2em",
            fontFamily: "mono",
            rounded: "sm",
            ...styles.value,
          }}
          {...attrs}
        >
          {() => getValidChildren(slots)}
        </chakra.code>
      )
    }
  },
})

export default CCode
