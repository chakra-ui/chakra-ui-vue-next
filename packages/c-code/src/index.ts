export { default as CCode } from './code'
