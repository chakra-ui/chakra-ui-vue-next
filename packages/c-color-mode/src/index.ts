export * from "./color-mode-provider"
export * from "./storage-manager"
export { setDataset } from "./color-mode.utils"
