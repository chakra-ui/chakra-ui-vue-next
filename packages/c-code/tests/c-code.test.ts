import { CCode } from '../src'
import { render, testA11y } from '../../test-utils/src'

const renderComponent = (props?: any) => {
  const base = {
    components: {
      CCode,
    },
    template: '<c-code>hello world</c-code>',
    ...props,
  }
  return render(base)
}

it('should have no a11y violations', async () => {
  await testA11y(renderComponent())
})

it('should render properly', () => {
  const { asFragment } = renderComponent()
  expect(asFragment()).toMatchSnapshot()
})
