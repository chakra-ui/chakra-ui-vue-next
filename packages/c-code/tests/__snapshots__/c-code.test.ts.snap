// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render properly 1`] = `
<DocumentFragment>
  <code
    class="chakra-lnsj14"
  >
    hello world
  </code>
  <code
    class="chakra-1ij4qum"
  >
     hello world, red 
  </code>
  <code
    class="chakra-l5fpz0"
  >
     hello world, green 
  </code>
  <code
    class="chakra-1ehvkaa"
  >
     hello world, yellow 
  </code>
  <code
    class="chakra-kugt48"
  >
     hello world, teal 
  </code>
</DocumentFragment>
`;
